.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,94 @@
+---
+# Full PR suite.  PRs need to pass this pipeline in order to be moved to the
+# merge queue, where they'll use the next rule as the branch-protection rule
+# for the final merge to the base branch.
+name: Pull Requests
+on:
+  pull_request:
+    branches: [main, "stable/*"]
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  # The preliminary stage should be small in both total runtime (including
+  # provisioning) and resources required.  About half of PR commits result in
+  # a CI failure, and over 90% of these are in linting, documention or a test
+  # failure that would affect _any_ OS or Python version.  The goal in the
+  # first stage is to catch the vast majority of failures with minimal cost.
+  lint-docs:
+    if: github.repository_owner == 'Qiskit'
+    name: Docs and Lint
+    uses: ./.github/workflows/lint_docs.yml
+  preliminary-tests:
+    name: Preliminary Tests
```

Comment:
If we've dropped the two stage approach and are going with maximum parallelism we should use consistent naming here. There is no preliminary or main. We also should just flatten the structure here so it's grouped by operating system/file. Right now the mix of the old and the new is confusing. Also none of the comments apply anymore.

Reply:
Fixed by a067d09 and 80c3268